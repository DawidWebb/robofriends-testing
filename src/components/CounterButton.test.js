import React from "react";
import { shallow } from "enzyme";
import CounterButton from "./CounterButton";

it("expoect to render CounterButton component", () => {
	const mockColor = `red`;

	expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
});

it("correctly increments the count", () => {
	const mockColor = `red`;
	const wrapper = shallow(<CounterButton color={mockColor} />);

	wrapper.find('[id="counter"]').simulate("click");
	expect(wrapper.state()).toEqual({ count: 1 });
	wrapper.find('[id="counter"]').simulate("click");
	wrapper.find('[id="counter"]').simulate("click");
	expect(wrapper.state()).toEqual({ count: 3 });
});
