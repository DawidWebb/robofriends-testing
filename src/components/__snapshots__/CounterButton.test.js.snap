// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expoect to render CounterButton component 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;

exports[`renders without crashing 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
